### ISSUE TITLE
tokens shouldn't be tokenized

### ISSUE THREAD

**Issue by alhuber1502** [marytts/marytts#281]
When feeding Mary a list of tokens using the Tokens input type, Mary shouldn't tokenize them.

**Comment by psibre** [comment]
Possibly related: [issue:353](http://mary.opendfki.de/ticket/353) on the old Trac.

**Comment by Munzey** [comment]
The problem seems to stem from mary's dependency on the freetts packages, namely its Utt2XMLBase module. Any conversion between two MARYXML types consists of firstly the target text being converted to a freetts Utterance,  followed by some (if any) kind of processing of these Utterances, before being converted back into MARYXML. Perhaps the problem could be solved by modifying the UTT2XMLBase module...

**Comment by psibre** [comment]
Another nail in the coffin of the FreeTTS dependencies, we should bump #228 up on our todo list...

**Comment by Munzey** [comment]
@psibre I was looking into this problem again as it is essentially the same issue we're having with trailing<'s> being tokenised. Ive been trying to debug in my fix-281 branch. The processing steps are as followed:
1. marytts is started up
2. modules needed to go from input to output are decided
3. first module is always XML2Utt[x] where x is some format such as words or tokens,etc. Here the sentence itself is converted to a FreeTTS utterance
4. some intermediate stage where a FreeTTS UtteranceProccessor is used
5. converted from an utterance and inserted back into a maryxml doc using some Utt2XML[y] class and then the other mary modules continue to process it.

In the case of going from tokens to phonemes, the modules used are: [marytts.language.en.XML2UttTokensEn@20c449e3, marytts.language.en.FreeTTSTokenToWords@36cee6e3, marytts.language.en.Utt2XMLWordsEn@49985d31, marytts.modules.OpenNLPPosTagger@320e61b6, marytts.modules.JPhonemiser@4e310d66]

For some reason, Mary chooses XML2UttTokensEn instead of XML2UttWordsEn. I dont see why it wouldn't just use that as I've already specified the tokens. There is also the issue that FreeTTSTokenToWords doesnt really provide anyway to edit the tokenising process as I dont have the source files. Here is the actual tokenising class javadoc: http://freetts.sourceforge.net/javadoc/com/sun/speech/freetts/en/us/TokenToWords.html

**Comment by Munzey** [comment]
I think this should actually be opened as a new issue regarding Mary's ModuleRegistry choosing the wrong modules for processing from input to output.

**Comment by Munzey** [comment]
scrap that last idea, going to start working on #228 now

**Comment by Munzey** [comment]
@alhuber1502 after investigating the issue further and reimplementing the `TOKEN->WORD` stage for english (which will be released with 5.2), I want to explain the processing design further:
your original question was whether there was a way to skip tokenization so that for a `TOKEN` input like:
```
<?xml version="1.0" encoding="UTF-8"?><maryxml xmlns="http://mary.dfki.de/2002/MaryXML" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="0.5" xml:lang="en-US">
<p>
<s>
<t>
bavaria's
</t>
<t>
fall
</t>
</s>
</p>
</maryxml>
```
could be output to the following stages while keeping this format and not splitting the 's away from the token into something like:
```
<?xml version="1.0" encoding="UTF-8"?><maryxml xmlns="http://mary.dfki.de/2002/MaryXML" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="0.5" xml:lang="en-US">
<p>
<s>
<t>
bavaria
</t>
<t>
's
</t>
<t>
fall
</t>
</s>
</p>
</maryxml>
```
The problem is that before a maryxml doc such as this can be passed on to the phonemization phase and further, we need to format the text into something that is pronouncable. This is where the `WORDS` mary data type is used. Following the tokenization module `JTokenizer`, the processing module `TokensToWords` ( but in 5.2+ will be called `Preprocess`) checks each of the tokens to see if they can be broken down further. This is very useful for formatting digits, dates, currency etc into spoken forms. e.g 24 -> twenty-four, 1st -> first , $20 -> twenty dollars

Something this module also tries to do is split tokens containing punctuation of any form into separate tokens, hence `<bavaria's>` becoming two tokens.
So as you can see disabling or skipping this phase is not really feasible, *however* in the new Preprocess implementation, a token containing punctuation is first checked to see if it is contained in the transcription dictionary. If that's the case then there is no need to split it into two tokens as when JPhonemiser attempts to process the token it will look it up in the lexicon and not have to rely on grapheme to phoneme rules.

Note: it just so happens that the new cmudict that we use for english contains a transcription for <bavaria's> so it doesnt get split up anymore :smiley:

Hope this helps!
